### ISSUE TITLE
Different results in the reports

### ISSUE THREAD

**Issue by diegolovison** [testng-team/testng#1830]
### TestNG Version
6.14.3

### Description
We have 3 tests: 
- FooTest:a
- FooTest:b
- BarTest:a

We have the following configuration in pom.xml
```xml
<excludedGroups>windows</excludedGroups>
```

In this case, 2 tests are supposed to be running and 1 should be skipped.

In the console we have:
```
Tests run: 2, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 0.176 sec - in TestSuite
```
In JUnitXMLReporter ( https://github.com/diegolovison/testng-reporter/blob/master/target/surefire-reports/Surefire%20suite/Surefire%20test.xml ) we have:
```xml
ignored="1"
<testcase name="a" ignored
```

In JUnitReportReporter ( https://github.com/diegolovison/testng-reporter/blob/master/target/surefire-reports/junitreports/TEST-com.github.diegolovison.testngreporter.FooTest.xml ) we have:
```xml
skipped="0"
```
No mention about FooTest:a

### Expected behavior
I would like to have the same result in console, JUnitXMLReporter and JUnitReportReporter

### Actual behavior
Different value in all reports

### Is the issue reproductible on runner?

- [ ] Shell
- [ X ] Maven
- [ ] Gradle
- [ ] Ant
- [ ] Eclipse
- [ ] IntelliJ
- [ ] NetBeans

### Test case sample
https://github.com/diegolovison/testng-reporter

**Comment by krmahadevan** [comment]
@diegolovison - Can you please check if this library that I built works for you ? https://github.com/RationaleEmotions/junitreport

**Comment by diegolovison** [comment]
Hi @krmahadevan

Well, no. I generated the report as you can see here https://github.com/diegolovison/testng-reporter/blob/rationale-emotions/target/surefire-reports/Surefire%20suite/Surefire%20test.xml
it does not show what I would like to see

The "junitreports" https://github.com/diegolovison/testng-reporter/tree/rationale-emotions/target/surefire-reports/junitreports shows what I am expecting.

What is your opinion?

**Comment by krmahadevan** [comment]
@diegolovison - I didn't quite understand. So are you saying that `junitreports` library that I shared works and solves your purpose?

**Comment by diegolovison** [comment]
@krmahadevan, no. In your report, where I can see that the method c in FooTest skipped?

```xml
  <testcase name="c" classname="com.github.diegolovison.testngreporter.FooTest">
    <skipped/>
  </testcase>
```

**Comment by krmahadevan** [comment]
@diegolovison - I am not sure if `skipped` is a recognized status in the JUnit world. My library refers to the JUnit.xsd available here https://github.com/windyroad/JUnit-Schema/blob/master/JUnit.xsd

**Comment by diegolovison** [comment]
Well, the info above related with `FooTest:c` comes from JUnitReportReporter as you can see here: https://github.com/diegolovison/testng-reporter/blob/rationale-emotions/target/surefire-reports/junitreports/TEST-com.github.diegolovison.testngreporter.FooTest.xml

What I would like to address in this issue is keep console, JUnitXMLReporter and JUnitReportReporter consistent. Today they have different results what cause confusion.

**Comment by krmahadevan** [comment]
@diegolovison - Let me see what I can do. Do you think you could help quickly share a sample that can be used to reproduce the problem ? That way its easy to get to the bottom of this issue.

**Comment by juherr** [comment]
Reviewed against the current built-in reporters: this is still relevant, and the inconsistency is visible in the current implementation.

`JUnitReportReporter` reads `suite.getExcludedMethods()`, but its mapping only keeps methods whose `enabled` flag is false. A method excluded by group while still enabled is therefore not emitted as an ignored/skipped testcase by that reporter, which is the exact category described here. I could not find a later regression or PR that normalizes this behavior across the console, `JUnitXMLReporter`, and `JUnitReportReporter`.

I did not rerun the historical Maven project. The minimal regression should use one enabled method excluded by group and assert the intended, consistent representation in the two built-in JUnit reporters before changing production code.

**Comment by diegolovison** [comment]
I just double-checked and I don't have more the repository available. Glad that you found a way to reproduce.
